src/routes/user.ts

```diff
@@ -1,29 +1,113 @@
+ *
+ * pageSize     = 15
+ * pageIndex    = 0
+ * sort         = last_name:asc,creation_date:desc
+ * roles        = Clinician,other
+ * dataUses     = Commercial use,other
+ *
+ */
+usersRouter.get(
+    '/search',
+    async (
+        req: Request<
+            {},
+            {},
+            {},
+            {
+                pageSize?: string;
+                pageIndex?: string;
+                sort?: string;
+                match?: string;
+                roles?: string;
+                dataUses?: string;
+                roleOptions?: string;
+                usageOptions?: string;
+            }
+        >,
+        res,
+        next,
+    ) => {
+        try {
+            const pageSize = parseInt(req.query.pageSize || '15');
+            const pageIndex = parseInt(req.query.pageIndex || '0');
+            const roles = req.query.roles ? req.query.roles.split(',') : [];
+            const dataUses = req.query.dataUses ? req.query.dataUses.split(',') : [];
+
+            let sorts: Order = [];
+            if (req.query.sort) {
+                sorts = req.query.sort.split(',').map((sortElement) => {
+                    const sortItems = sortElement.split(':');
+                    return [sortItems[0], sortItems[1].toUpperCase()];
+                });
+            }
+
+            const roleOptions = req.query.roleOptions ? req.query.roleOptions.split(',') : [];
+            const usageOptions = req.query.usageOptions ? req.query.usageOptions.split(',') : [];
 
-        const result = await searchUsers(pageSize, pageIndex);
+            if (!roleOptions.length || !roleOptions.length) {
+                throw createHttpError(StatusCodes.BAD_REQUEST, 'roleOptions and usageOptions array must be provided.');
+            }
+
+            const result = await searchUsers({
+                pageSize,
+                pageIndex,
+                sorts,
+                match: req.query.match,
+                roles,
+                dataUses,
+                roleOptions,
+                usageOptions,
+            });
+            res.status(StatusCodes.OK).send(result);
+        } catch (e) {
+            next(e);
+        }
+    },
+);
+
+usersRouter.get('/image/presigned', async (req, res, next) => {
+    try {
+        const keycloak_id = req['kauth']?.grant?.access_token?.content?.sub;
+        const result = await getProfileImageUploadPresignedUrl(keycloak_id);
         res.status(StatusCodes.OK).send(result);
     } catch (e) {
         next(e);
     }
 });
 
-usersRouter.get('/', async (req, res, next) => {
+usersRouter.get('/:id?', async (req, res, next) => {
     try {
         const keycloak_id = req['kauth']?.grant?.access_token?.content?.sub;
-        const result = await getUserById(keycloak_id);
+        const requestKeycloakId = req.params.id;
+
+        const result = await getUserById(
+            requestKeycloakId ?? keycloak_id,
+            requestKeycloakId ? requestKeycloakId === keycloak_id : true,
```

Comment:
could that be simplified?

Reply:
Encore une fois ca vient d'include puis le but est de prendre le moins de risque possible.
Mais `requestKeycloakId ? requestKeycloakId === keycloak_id : true` en fin de compte peut pas etre bcp plus simplifier.
 Mettre seulement `requestKeycloakId === keycloak_id` passerait `requestKeycloakId` a false si il existe pas. Alors que la premiere le passe a true.
Olivier avait donc ses raison pour coder ce ternaire.